server/webapp/WEB-INF/rails.new/app/assets/new_javascripts/models/template_configs/template.js

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2016 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+define(['mithril', 'lodash', 'string-plus', 'helpers/mrequest', 'js-routes', 'models/model_mixins',
+    'models/validatable_mixin', 'models/pipeline_configs/stages', 'models/pipeline_configs/jobs'],
+  function (m, _, s, mrequest, Routes, Mixins, Validatable, Stages, Jobs) {
+
+    Template = function (data) {
+      Validatable.call(this, data);
+
+      this.name          = m.prop(s.defaultToIfBlank(data.name, ''));
+      this.authorization = m.prop(s.defaultToIfBlank(data.authorization, new Template.Authorization({})));
+      this.stages        = s.collectionToJSON(m.prop(s.defaultToIfBlank(data.stages, new Stages())));
+
+      this.validatePresenceOf('name');
+      this.validateAssociated('stages');
+
+      this.update = function (etag, extract) {
+        var self = this;
+
+        var config = function (xhr) {
```

Comment:
All you wanted is to add etag to the xhr, you will get xhr object in config itself. Please look at https://github.com/gocd/gocd/blob/master/server/webapp/WEB-INF/rails.new/app/assets/new_javascripts/models/elastic_profiles/elastic_profiles.js#L72